tools/adventure-pack/goodies/java/UnionFind.java

```diff
@@ -0,0 +1,24 @@
+class UnionFind {
+    
+    private int[] parent;
+    
+    public UnionFind(int n){
+        parent = new int[n];
+        for(int i = 0; i < n; i++){
+            parent[i] = i;
+        }
+    }
+    
+    public void union(int a, int b){
+        int pa = find(a);
+        int pb = find(b);
+        
+        parent[pa] = parent[pb];
+    }
+    
+    public int find(int a){
+        if(parent[a] == a) return a;
```

Comment:
Out of curiosity, is there a Java linter you'd recommend? I'd probably want to enforce that if statements and other blocks always use braces. (But I also don't think this should be enforced manually.)

Reply:
I installed clang-format and ran that. I'm open to a different one if you have a preference.